test/util/validation/validate_service.go

```diff
@@ -233,3 +244,117 @@ func ValidateServicePods(ctx context.Context, k8sClient client.Client, service *
 		return nil
 	}).Should(gomega.Succeed())
 }
+
+type CheckServiceAvailableFunc func(int) error
+
+func ValidateServiceAvaliable(ctx context.Context, k8sClient client.Client, cfg *rest.Config, service *inferenceapi.Service, check CheckServiceAvailableFunc) {
+	gomega.Eventually(func() error {
+		pods := corev1.PodList{}
+		podSelector := client.MatchingLabels(map[string]string{
+			lws.SetNameLabelKey: service.Name,
+		})
+		if err := k8sClient.List(ctx, &pods, podSelector, client.InNamespace(service.Namespace)); err != nil {
+			return err
+		}
+		if len(pods.Items) != int(*service.Spec.Replicas)*int(*service.Spec.WorkloadTemplate.Size) {
+			return fmt.Errorf("pods number not right, want: %d, got: %d", int(*service.Spec.Replicas)*int(*service.Spec.WorkloadTemplate.Size), len(pods.Items))
+		}
+
+		var targetPod *corev1.Pod
+		for i := range pods.Items {
+			if pods.Items[i].Status.Phase == corev1.PodRunning {
+				targetPod = &pods.Items[i]
+				break
+			}
+		}
+
+		if targetPod == nil {
+			return fmt.Errorf("no running pods found for service %s", service.Name)
+		}
+
+		portForwardK8sClient, err := kubernetes.NewForConfig(cfg)
+		if err != nil {
+			return fmt.Errorf("init port forward client failed: %w", err)
+		}
+
+		targetPort := targetPod.Spec.Containers[0].Ports[0].ContainerPort
+		localPort := 8080
```

Comment:
if we hardcoded the localport, let's just remove the args in `CheckServiceAvaliable`, we can add back once needed.